pkg/reconciler/instances/istio/reset/data/gatherer_test.go

```diff
@@ -91,7 +94,7 @@ func Test_Gatherer_GetPodsWithDifferentImage(t *testing.T) {
 		// then
 		require.Equal(t, podsWithDifferentImage.Items, expected.Items)
 		require.NotEmpty(t, podsWithDifferentImage.Items)
-		require.Len(t, podsWithDifferentImage.Items, 1)
+		require.Len(t, podsWithDifferentImage.Items, 2)
```

Comment:
What about validating which exactly pods are included in the list? Would it be possible to validate if the correct ones that you truly expect are here?